controllers/managedocs_controller.go

```diff
@@ -483,7 +483,11 @@ func (r *ManagedOCSReconciler) reconcilePhases() (reconcile.Result, error) {
 		}
 
 		// Reconcile the different resources
-		if err := r.reconcileRookCephOperatorConfig(); err != nil {
+		if err := r.reconcileConvergedRookCephOperatorConfig(); err != nil {
+			return ctrl.Result{}, err
+		}
+		// Reconciles only for provider deployment type
+		if err := r.reconcileProviderRookCephOperatorConfig(); err != nil {
 			return ctrl.Result{}, err
 		}
```

Comment:
The code for both helpers seems almost identical and is not predicated to diverge any further in the future. 
Can we just solve it with a simple if inside the original `reconcileRookCephOperatorConfig` func?